pyanaconda/modules/boss/install_manager/install_manager.py

```diff
@@ -0,0 +1,264 @@
+        self._error_raised_signal = Signal()
+
+        self._subscriptions = []
+
+    @property
+    def install_running_signal(self):
+        """Signal when installation starts/stops."""
+        return self._install_running_signal
+
+    @property
+    def task_changed_signal(self):
+        """Signal when installation task changed."""
+        return self._task_changed_signal
+
+    @property
+    def progress_changed_signal(self):
+        """Signal when progress changed."""
+        return self._progress_changed_signal
+
+    @property
+    def progress_changed_float_signal(self):
+        """Signal when progress in float changed."""
+        return self._progress_changed_float_signal
+
+    @property
+    def error_raised_signal(self):
+        """Signal which will be emitted when error raised during installation."""
+        return self._error_raised_signal
+
+    @property
+    def available_modules(self):
+        """Get available modules which will be used for installation."""
+        return self._modules
+
+    @available_modules.setter
+    def available_modules(self, modules):
+        """Set available modules which will be used for installation.
+
+        :param modules: Modules list.
+        :type modules: list
+        """
+        self._modules = modules
+
+    def start_installation(self):
+        """Start the installation."""
+        self._collect_tasks()
+
+        self._sum_steps_count()
+        self._disconnect_task()
+        self._tasks_done_step = 0
+        self._installation_terminated = False
+
+        if self._tasks:
+            self._actual_task = self._tasks.pop()
+            self._install_running_signal.emit(True)
+            self._run_task()
+
+    def _collect_tasks(self):
+        self._tasks.clear()
+
+        if not self._modules:
+            log.error("Starting installation without available modules.")
+
+        for module_service in self._modules:
+            # FIXME: This is just a temporary solution.
+            module_object = DBus.get_proxy(module_service, auto_object_path(module_service))
+
+            tasks = module_object.AvailableTasks()
+            for task in tasks:
+                log.debug("Getting task %s from module %s", task[TASK_NAME], module_service)
+                task_proxy = DBus.get_proxy(module_service, task[TASK_PATH])
+                self._tasks.add(task_proxy)
+
+    def _sum_steps_count(self):
+        self._step_sum = 0
+        for task in self._tasks:
+            self._step_sum += task.ProgressStepsCount
+
+    def _run_task(self):
+        if self._installation_terminated:
+            log.debug("Don't run another task. The installation was terminated.")
+            return
+
+        task_name = self._actual_task.Name
+
+        log.debug("Running installation task %s", task_name)
+        self._disconnect_task()
+        self._connect_task()
+        self._task_changed_signal.emit(task_name)
+        self._actual_task.Run()
```

Comment:
Shouldn't be this done asynchronously? I don't think you are able to do anything else while the task is running.

Reply:
Ok, this is not true, because the Run method only starts the task. I am not sure if it shouldn't be Start method.

Reply:
Could be renamed, I don't mind.